src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,492 @@
+``FindByValue``. Here are a few observations:
+
+#. The method that you write should not contain duplicate jobs. So, for
+   example, if a listing has position type “Web - Front End” and name
+   “Front end web dev” then searching for “web” should not include the
+   listing twice.
+#. As with ``PrintJobs``, you should write your code in a way that if a
+   new column is added to the data, your code will automatically search
+   the new column as well.
+#. You should NOT write code that calls ``FindByColumnAndValue`` once
+   for each column. Rather, utilize loops and collection methods as you
+   did above.
+#. You *should*, on the other hand, read and understand
+   ``FindByColumnAndValue``, since your code will look similar in some
+   ways.
+
+You’ll need to call ``FindByValue`` from somewhere in ``Main``. We’ll
+leave it up to you to find where. You might have noticed that when you
+try to search all columns using the app, a message is printed, so that
+is a good clue to help you find where to place this new method call.
+
+Once you find a spot to call your new method, run the program again to
+test your method call.
+
+Make Search Methods Case-Insensitive
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You’ve completed your first two tasks!
+
+Let's assume you demonstrated the updated application for the Company Team, and
+they noticed a feature that could be improved. When searching for jobs with
+the skill ``"JavaScript"``, some results were missing (e.g. the Watchtower
+Security job on line 31 of the CSV file). The search methods turn out to be
+case-sensitive, so they treat ``"JavaScript"`` and ``"Javascript"`` as different
+strings.
+
+The Company Team strongly requested that this needs to be fixed, and of course,
+you told them that you are up to the task.
+
+Here are some questions to ask yourself as you get started:
+
+#. Which methods are called when searching?
+#. How is the user’s search string compared against the values of fields of the
+   job ``Dictionary`` objects?
+#. How can you make this comparison in a way that effectively ignores the case
+   of the strings?
+#. How can you do this *without* altering the capitalization of the items in
+   ``AllJobs`` so that the data gets printed out the same way that it appears
+   in ``job_data.csv``?
+
+Once you have an idea for how to approach this, you’ll likely benefit from your 
+favorite search engine to find out exactly how to do this in C#. You may also find 
+it useful to review :ref:`string-methods`.
+
+When this task is completed, you’re done!
+
+Sanity Check
+------------
+
+Before submitting, make sure that your application:
+
+#. Prints each field of a job when using search functionality, and when
+   listing all columns. If there are no search results, a descriptive
+   message is displayed.
+#. Allows the user to search for a string across all columns.
+#. Returns case-insensitive results.
+
+Solution Demo
+-------------
+
+Watch a demo of a working solution.
```

Comment:
I would add a note that the solution is in Java so that it doesn't confuse students.

Reply:
huh? this video is for the c# solutions. do you see something different?

Reply:
I clicked on the wrong vid on Youtube. My bad!